### PR TITLE
feat: check conditional requirements on translations fields

### PR DESCRIPTION
**Pull request by lionel-nj**
**Summary:**

This PR provides support to verify conditional requirements on GTFS file `translations.txt`.

**New notices:** 
- TranslationFieldValueDefinedNotice
- TranslationFieldValueNotDefinedNotice
- TranslationRecordIdDefinedNotice
- TranslationRecordIdNotDefinedNotice
- TranslationRecordSubIdDefinedNotice
- TranslationRecordSubIdNotDefinedNotice

**Expected behavior:** 

A notice should be generated and added to the `NoticeContainer` if: 
- table_name is feed_info and:
  - record_id is defined
  - record_sub_id is defined
  - field_value is defined
 - table_name is stop_times, record_id is defined and record_sub_id is not defined
 - field_value is defined and:
   - record_id is defined
   - record_sub_id is defined
  - field_value is empty
    - record_id is not defined
 - field_value and record_sub_id are defined
 - field_value and record_id are defined 
 - record_id is empty and field value is not defined
 
http://gtfs.org/reference/static#translationstxt

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

**Comment by lionel-nj** [comment]
@barbeau Should `GtfsTranslation.tableName()` return and enum value instead of String values? This field is defined by the spec as an enum and the speficiation says:

> Allowed values are: agency, stops, routes, trips, stop_times, pathways, levels, feed_info and attributions (do not include the .txt file extension)

**Review by maximearmstrong — CHANGES_REQUESTED** [review]

**Review by barbeau — COMMENTED** [review]
@lionel-nj Here's what I have so far, although I haven't looked at the implementations yet.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TranslationConditionalRequirementsValidator.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+@GtfsValidator
+public class TranslationConditionalRequirementsValidator extends FileValidator {
+
+  private static final String TXT_SUFFIX = ".txt";
+  @Inject GtfsTranslationTableContainer translationTable;
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    for (GtfsTranslation translation : translationTable.getEntities()) {
+      if (translation
```
Comment (aababilov) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TranslationConditionalRequirementsValidator.java:73]: Unfortunately, that approach does not scale well. GTFS spec defines a lot of files and users may add their extensions with new fields and fields.

We need changes to the validator core. The annotation processor should provide information about the schema and validator classes may receive that as a part of validation context.

I can make the necessary changes to the core when I come back after 15 March.

**Comment by lionel-nj** [comment]
Closing for now, will resume implementation as part of v2.1.

**Pull request closed without merging**